pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/PicklistQuestionDao.java

```diff
@@ -105,15 +108,69 @@ default void insertGroup(long questionId, PicklistGroupDef group, int order, lon
+        List<Template> templateList = Stream.of(
+                options.stream().map(o -> o.getOptionLabelTemplate()),
+                options.stream().map(o -> o.getTooltipTemplate()),
+                options.stream().map(o -> o.getDetailLabelTemplate()),
+                options.stream().map(o -> o.getNestedOptionsLabelTemplate())
+        ).flatMap(i -> i).collect(toList());
+
+        Long[] templateIds = getTemplateDao().insertTemplates(templateList, revisionId);
+        List<Long> templateIdList = Arrays.asList(templateIds);
+
+        JdbiPicklistOption jdbiOption = getJdbiPicklistOption();
+
+        // batch enter all the options
+        long[] optionIds = jdbiOption.insert(questionId,
+                options.stream().map(o -> o.getStableId()).iterator(),
+                templateIdList.listIterator(),
+                templateIdList.listIterator(options.size()),
+                templateIdList.listIterator(2 * options.size()),
+                options.stream().map(o -> o.isDetailsAllowed()).iterator(),
+                options.stream().map(o -> o.isExclusive()).iterator(),
+                Stream.iterate(0, i -> i + DISPLAY_ORDER_GAP).iterator(),
+                revisionId,
+                templateIdList.listIterator(3 * options.size()));
+
         int displayOrder = 0;
+        int j = 0;
         for (PicklistOptionDef option : options) {
             displayOrder += DISPLAY_ORDER_GAP;
-            long optionId = insertOption(questionId, option, displayOrder, revisionId);
-            optionIds.add(optionId);
+            option.setOptionId(optionIds[j++]);
+
+            insertNestedOptions(questionId, option, displayOrder, revisionId);
         }
+
         LOG.info("Inserted {} picklist options for picklist question id {}", options.size(), questionId);
-        return optionIds;
+        return options.stream().map(o -> o.getOptionId()).collect(toList());
+    }
+
+    default void insertNestedOptions(long questionId, PicklistOptionDef option,
+                                     int displayOrder, long revisionId) {
+
+        if (CollectionUtils.isNotEmpty(option.getNestedOptions())) {
+            int subOptionDisplayOrder = displayOrder;
+            List<Long> nestedOptionIds = new ArrayList<>();
+            for (PicklistOptionDef nestedOption : option.getNestedOptions()) {
+                subOptionDisplayOrder += DISPLAY_ORDER_GAP;
+                long nestedOptionId = insertOption(questionId, nestedOption, subOptionDisplayOrder, revisionId);
+                nestedOptionIds.add(nestedOptionId);
+            }
+            //now populate nested options join table
+            getJdbiPicklistOption().bulkInsertNestedOptions(option.getOptionId(), nestedOptionIds);
+            LOG.info("Inserted {} nested options for picklist option: {} optionId: {}",
+                    nestedOptionIds.size(), option.getStableId(), option.getOptionId());
```

Comment:
There was existing batch ("bulk") inserting going on, though probably benefit is limited (not many questions have nested options AND in number significant to make visible speedup difference).